### PR TITLE
Restrict the cuDNN warning not to print more than once

### LINKED ISSUE #1453: Stop printing cuDNN warning more than once
Users do not need multiple cuDNN warnings. https://github.com/pfnet/chainer/blob/71eef3e90baea61d3140b75741520515f8882f9d/chainer/cuda.py#L86

### PR DESCRIPTION
**Pull request by gwtnb**
Fix #1453

### CODE REVIEW

File: chainer/cuda.py [hunk 1]
```diff
@@ -82,11 +82,13 @@ def check_cuda_available():
                '(see https://github.com/pfnet/chainer#installation).')
         msg += str(_resolution_error)
         raise RuntimeError(msg)
-    if not cudnn_enabled:
+    if (not cudnn_enabled and
+            not getattr(check_cuda_available, '_already_warned', False)):
```
Comment (okuta) [inline chainer/cuda.py]: Is it possible not to use `getattr`?

Comment (gwtnb) [inline chainer/cuda.py]: It is possible. Could you tell me why `getattr` is undesirable?

Comment (okuta) [inline chainer/cuda.py]: `check_cuda_available._already_warned` is simple, isn't it?

Comment (gwtnb) [inline chainer/cuda.py]: To eliminate `getattr`, `check_cuda_available._already_warned = False` should be executed before any call of `check_cuda_available`. Is it simple?

Alternatively, I can use a global variable instead, but it seems unclear for a reader. `_already_warned` should be belong to `check_cuda_available` for readability.

Comment (okuta) [inline chainer/cuda.py]: OK. I understood.

**Comment by okuta** [comment]
LGTM

**Pull request merged**